### PR TITLE
[RFC] Style and spelling

### PR DESCRIPTION
**Pull request by SibrenVasse**
Made some style and spelling changes. 
- Changed apostrophe in strings from \` to '
- Where possible, reduced length of strings to better fit on screen
- Small indentation changes
- Few language/spelling changes

What do you think?

### CODE REVIEW

File: bootiso [hunk 1]
```diff
@@ -463,15 +463,17 @@ configureLabel() {
...
-		echowarn "Unexpected partition type \`$partitionType'.\\n$openTicketMessage"
+		echowarn "Unexpected partition type '$partitionType'."
+		echowarn "$openTicketMessage"
 		;;
 	esac
 	# Fallback to "BOOTISO"
 	partitionLabel=${partitionLabel:-"BOOTISO"}
 	if [ -z "${userVars['label']}" ]; then
-		echogood "Partition label automatically set to \`$partitionLabel'.\\nYou can explicitly set label with \`-L, --label' option."
+		echogood "Partition label automatically set to '$partitionLabel'."
```
Comment (jsamr) [inline bootiso]: echogood prints `bootiso: ...`. That's why I put a `\n` character.

File: bootiso [hunk 2]
```diff
@@ -480,26 +482,28 @@ configureLabel() {
...
+		echowarn "Command '$1' not found! Should be in package '${commandPackages["$1"]}'."
 		if [ ! -z "$pkgmgr" ]; then
-			read -r -n1 -p "Attempt installation? (y/n)>" answer
+			read -r -n1 -p "Attempt installation? (y/n)> " answer
 			echo
 			case $answer in
 			y | Y)
 				if ! $pkgmgr "${commandPackages["$1"]}"; then
-					failAndExit "Installation of dependency \`$1' failed.\\nPerhaps this dependency has a slightly different name in your distribution.\\nFind it and install manually."
+					echoerr "Installation of dependency '$1' failed."
```
Comment (jsamr) [inline bootiso]: Same as above, echoerr will print `bootiso: ...`

File: bootiso [hunk 3]
```diff
@@ -546,7 +550,9 @@ listDevicesTable() {
 		$lsblkCmd | grep --color=never -P "^($(joinBy '|' "${devicesList[@]}"))" | sed 's/^/         /'
 		return 0
 	else
-		echowarn "Couldn't find any USB drive in your system.\\nIf any is physically plugged in, it's likely that it has been ejected and should be plugged-in again to be discoverable.\\nYou can check the availability of USB drives with \`$scriptName -l' command."
+		echowarn "Couldn't find any USB drives on your system."
+		echowarn "If one is physically plugged in, it's likely that it has been ejected and should be reconnected."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 4]
```diff
@@ -821,13 +828,14 @@ assertDeviceIsOK() {
...
-		failDevice "The selected device \`$device' does not exists."
+		failDevice "The selected device '$device' does not exist."
 	fi
 	if [ ! -b "$device" ]; then
-		failDevice "The selected device \`$device' is not a valid block file."
+		failDevice "The selected device '$device' is not a valid block file."
 	fi
 	if [ ! -d "/sys/block/$(basename "$device")" ] || ! deviceIsDisk "$device"; then
-		failAndExit "The selected device \`$device' is either unmounted or not a disk (might be a partition or loop).\\nSelect a disk instead or replug the USB device.\\nYou can check the availability of USB drives with \`$scriptName -l' command."
+		echoerr "The selected device '$device' is either unmounted or not a disk (might be a partition or loop)."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 5]
```diff
@@ -1070,10 +1078,11 @@ installSyslinuxVersion() {
 		status=$(curl -sLIo /dev/null -w "%{http_code}" "$rootURL")
 		if [ "$status" != 200 ]; then
 			if [ "$status" == 000 ]; then
-				failAndExit "You don't seem to have an internet connexion.\\nPlease try again later or use \`--local-bootloader' option to force usage of local SYSLINUX version."
+				echoerr "You don't seem to have an internet connection."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 6]
```diff
@@ -1302,24 +1313,25 @@ checkUserVars() {
 		assertDeviceIsOK "${userVars['device']}"
 	fi
 	if [ ! -z "${userVars['remote-bootloader']}" ] && [[ ! "${userVars['remote-bootloader']}" =~ ^[0-9]+\.[0-9]+$ ]]; then
-		failAndExit "Remote bootloader version \`${userVars['remote-bootloader']}' set with \`--remote-bootloader' doesn't follow MAJOR.MINOR pattern.\\nValid examples are 4.10, 6.02"
+		echoerr "Remote bootloader version '${userVars['remote-bootloader']}' set with '--remote-bootloader' doesn't follow MAJOR.MINOR pattern."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 7]
```diff
@@ -1511,7 +1525,8 @@ inspectISOBootCapabilities() {
 	typeset syslinuxCompatible=false
 	if [ "${isoInspection['supportsEFIBoot']}" == "true" ]; then
 		if [ "$partitionType" != "vfat" ]; then
-			echowarn "Found UEFI boot capabilities but you selected \`$partitionType' type, which is not compatible with UEFI boot.\\nBe warned that only legacy boot will work, if any."
+			echowarn "Found UEFI boot capabilities but you selected '$partitionType' type, which is not compatible with UEFI boot."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 8]
```diff
@@ -1530,22 +1545,24 @@ inspectISOBootCapabilities() {
 		echogood "A SYSLINUX booloader will be installed on your USB."
 	fi
 	if [ "$syslinuxCompatible" == false ] && [ "$uefiCompatible" == false ]; then
-		failAndExit "The selected ISO is not hybrid, doesn't supports UEFI nor legacy booting with SYSLINUX.\\nTherefore, it cannot result in any successful booting with $scriptName.\\nConsider following the documentation provided with this ISO file."
+		echoerr "The selected ISO is not hybrid, doesn't supports UEFI nor legacy booting with SYSLINUX."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 9]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
 	if [ ! -z "${isoInspection['syslinuxVer']}" ]; then
 		syslinux+=" ${isoInspection['syslinuxVer']}"
 	fi
-	echoinfo "Reporting \`$selectedIsoFile' boot capabilities:"
+	echoinfo "Reporting '$selectedIsoFile' boot capabilities:"
 	if [ "$isHybrid" == false ] && [ "$syslinuxCompatible" == false ] && [ "$uefiCompatible" == false ]; then
-		echoerr "The selected ISO is not hybrid, doesn't supports UEFI nor legacy BIOS booting with SYSLINUX.\\nIt cannot result in any successful booting with $scriptName."
+		echoerr "The selected ISO is not hybrid, doesn't support UEFI or legacy BIOS booting with SYSLINUX."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 10]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
+		echoerr "It cannot result in any successful booting with $scriptName."
 	elif [ "$isHybrid" == false ] && [ "$syslinuxCompatible" == true ] && [ "$uefiCompatible" == false ]; then
-		echogood "The selected ISO is not hybrid, but supports legacy BIOS booting with $syslinux.\\nIt should result in successfull booting with $scriptName in [install-auto] and [install-mount-rsync] modes with any PC BIOS."
+		echogood "The selected ISO is not hybrid, but supports legacy BIOS booting with $syslinux."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 11]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
+		echogood "It should boot with $scriptName in [install-auto] and [install-mount-rsync] modes with any PC BIOS."
 	elif [ "$isHybrid" == false ] && [ "$syslinuxCompatible" == false ] && [ "$uefiCompatible" == true ]; then
-		echogood "The selected ISO is not hybrid, but supports UEFI boot.\\nIt will result in successfull booting with $scriptName in [install-auto] and [install-mount-rsync] modes with modern UEFI-enabled PC BIOS."
+		echogood "The selected ISO is not hybrid, but supports UEFI boot."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 12]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
+		echogood "It will boot with $scriptName in [install-auto] and [install-mount-rsync] modes with modern UEFI-enabled PC BIOS."
 	elif [ "$isHybrid" == false ] && [ "$syslinuxCompatible" == true ] && [ "$uefiCompatible" == true ]; then
-		echogood "The selected ISO is not hybrid, but supports legacy BIOS booting with $syslinux and UEFI boot.\\nIt will result in successfull booting with $scriptName in [install-auto] and [install-mount-rsync] modes with any PC BIOS."
+		echogood "The selected ISO is not hybrid, but supports legacy BIOS booting with $syslinux and UEFI boot."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 13]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
+		echogood "It will boot with $scriptName in [install-auto] and [install-mount-rsync] modes with any PC BIOS."
 	elif [ "$isHybrid" == true ] && [ "$syslinuxCompatible" == false ] && [ "$uefiCompatible" == false ]; then
-		echowarn "The selected ISO is hybrid, but doesn't support UEFI nor SYSLINUX legacy BIOS bootloader.\\nIt should result in successfull booting with $scriptName in [install-auto] and [install-dd] modes, while $scriptName is not aware of its booting scheme."
+		echowarn "The selected ISO is hybrid, but doesn't support UEFI or SYSLINUX legacy BIOS bootloader."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 14]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
+			echogood "SYSLINUX version in ISO file matches local version (${isoInspection['syslinuxVer']})."
 		elif compute "${localSyslinuxVersion%.*} == ${isoInspection['syslinuxVer']%.*}"; then
-			echoinfo "However, SYSLINUX version (${isoInspection['syslinuxVer']}) in ISO file doesn't match minor part of local version ($localSyslinuxVersion) which should not cause any problems."
+			echoinfo "SYSLINUX version (${isoInspection['syslinuxVer']}) in the ISO file doesn't match the minor part of local version ($localSyslinuxVersion), which should not cause any problems."
 		else
-			echowarn "However, SYSLINUX version (${isoInspection['syslinuxVer']}) in ISO file doesn't match major part of local version ($localSyslinuxVersion).\\n$scriptName will try to download and execute this version from kernel.org, unless given the option \`--local-bootloader'.\\nIn last resort if that fails, it will attempt installation with local version of SYSLINUX."
+			echowarn "SYSLINUX version (${isoInspection['syslinuxVer']}) in the ISO file doesn't match the major part of local version ($localSyslinuxVersion)."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 15]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
 	elif [ "$isHybrid" == true ] && [ "$syslinuxCompatible" == true ] && [ "$uefiCompatible" == true ]; then
-		echogood "The selected ISO is hybrid and supports legacy BIOS booting with $syslinux along with UEFI boot.\\nIt will result in successfull booting with $scriptName in [install-auto] and [install-dd] modes with any PC BIOS."
+		echogood "The selected ISO is hybrid and supports legacy BIOS booting with $syslinux along with UEFI boot."
+		echogood "It will boot with $scriptName in [install-auto] and [install-dd] modes with any PC BIOS."
 	fi
 	typeset localSyslinuxVersion=$(syslinux --version |& grep -oP '(\d+\.\d+)')
 	if [ "$syslinuxCompatible" == true ] && [ ! -z "${isoInspection['syslinuxVer']}" ] && [ "$isHybrid" == false ]; then
 		if compute "$localSyslinuxVersion == ${isoInspection['syslinuxVer']}"; then
-			echogood "Furthermore, SYSLINUX version in ISO file matches local version (${isoInspection['syslinuxVer']})."
+			echogood "SYSLINUX version in ISO file matches local version (${isoInspection['syslinuxVer']})."
```
Comment (jsamr) [inline bootiso]: 'Furthermore' intended (following previous sentence)

File: bootiso [hunk 16]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
 		elif compute "${localSyslinuxVersion%.*} == ${isoInspection['syslinuxVer']%.*}"; then
-			echoinfo "However, SYSLINUX version (${isoInspection['syslinuxVer']}) in ISO file doesn't match minor part of local version ($localSyslinuxVersion) which should not cause any problems."
+			echoinfo "SYSLINUX version (${isoInspection['syslinuxVer']}) in the ISO file doesn't match the minor part of local version ($localSyslinuxVersion), which should not cause any problems."
```
Comment (jsamr) [inline bootiso]: 'However' intended (following previous sentence)

File: bootiso [hunk 17]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
+		echogood "The selected ISO is hybrid and supports legacy BIOS booting with $syslinux."
+		echogood "It will boot with $scriptName in [install-auto] and [install-dd] modes with any PC BIOS."
 	elif [ "$isHybrid" == true ] && [ "$syslinuxCompatible" == false ] && [ "$uefiCompatible" == true ]; then
-		echogood "The selected ISO is hybrid and supports UEFI boot.\\nIt will result in successfull booting with $scriptName in [install-auto] and [install-dd] modes with modern UEFI-enabled PC BIOS."
+		echogood "The selected ISO is hybrid and supports UEFI boot."
+		echogood "It will boot with $scriptName in [install-auto] and [install-dd] modes with modern UEFI-enabled PC BIOS."
 	elif [ "$isHybrid" == true ] && [ "$syslinuxCompatible" == true ] && [ "$uefiCompatible" == true ]; then
-		echogood "The selected ISO is hybrid and supports legacy BIOS booting with $syslinux along with UEFI boot.\\nIt will result in successfull booting with $scriptName in [install-auto] and [install-dd] modes with any PC BIOS."
+		echogood "The selected ISO is hybrid and supports legacy BIOS booting with $syslinux along with UEFI boot."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 18]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
+		echowarn "The selected ISO is hybrid, but doesn't support UEFI or SYSLINUX legacy BIOS bootloader."
+		echowarn "It should boot with $scriptName in [install-auto] and [install-dd] modes, but $scriptName is not aware of its booting scheme."
 	elif [ "$isHybrid" == true ] && [ "$syslinuxCompatible" == true ] && [ "$uefiCompatible" == false ]; then
-		echogood "The selected ISO is hybrid and supports legacy BIOS booting with $syslinux.\\nIt will result in successfull booting with $scriptName in [install-auto] and [install-dd] modes with any PC BIOS."
+		echogood "The selected ISO is hybrid and supports legacy BIOS booting with $syslinux."
+		echogood "It will boot with $scriptName in [install-auto] and [install-dd] modes with any PC BIOS."
 	elif [ "$isHybrid" == true ] && [ "$syslinuxCompatible" == false ] && [ "$uefiCompatible" == true ]; then
-		echogood "The selected ISO is hybrid and supports UEFI boot.\\nIt will result in successfull booting with $scriptName in [install-auto] and [install-dd] modes with modern UEFI-enabled PC BIOS."
+		echogood "The selected ISO is hybrid and supports UEFI boot."
```
Comment (jsamr) [inline bootiso]: idem

File: bootiso [hunk 19]
```diff
@@ -1560,32 +1577,42 @@ printISOBootCapabilities() {
...
+		echogood "It will boot with $scriptName in [install-auto] and [install-mount-rsync] modes with any PC BIOS."
 	elif [ "$isHybrid" == true ] && [ "$syslinuxCompatible" == false ] && [ "$uefiCompatible" == false ]; then
-		echowarn "The selected ISO is hybrid, but doesn't support UEFI nor SYSLINUX legacy BIOS bootloader.\\nIt should result in successfull booting with $scriptName in [install-auto] and [install-dd] modes, while $scriptName is not aware of its booting scheme."
+		echowarn "The selected ISO is hybrid, but doesn't support UEFI or SYSLINUX legacy BIOS bootloader."
+		echowarn "It should boot with $scriptName in [install-auto] and [install-dd] modes, but $scriptName is not aware of its booting scheme."
 	elif [ "$isHybrid" == true ] && [ "$syslinuxCompatible" == true ] && [ "$uefiCompatible" == false ]; then
-		echogood "The selected ISO is hybrid and supports legacy BIOS booting with $syslinux.\\nIt will result in successfull booting with $scriptName in [install-auto] and [install-dd] modes with any PC BIOS."
+		echogood "The selected ISO is hybrid and supports legacy BIOS booting with $syslinux."
```
Comment (jsamr) [inline bootiso]: idem

**Review by jsamr — CHANGES_REQUESTED** [review]
@SibrenVasse First, thanks a lot for your time, and the typos !

Two remarks: 

- As per the backtick + single quote style, [it is a legacy and tradition in limited charsets](https://english.stackexchange.com/questions/17695/any-reference-on-the-usage-of-a-backtick-and-single-quotation-mark-like-this/18669) which I was honouring. However, I'm OK with this style change.
- You often split `echoerr`, `echogood` and `echoinfo` in multiple call, I guess not to hit a line width threshold. The issue is the code is not equivalent:

``` bash
echogood "foo bar"
echogood "foo zed"
```
prints:
```
bootiso: foo bar
bootiso: foo zed
```
whereas 
``` bash
echogood "foo bar\\nfoo zed"
```
prints:
``` bash
bootiso: foo bar
         foo zed
```
which is intended to group sentences by message.

So please revert those changes and I'll accept this PR :-)

**Comment by SibrenVasse** [comment]
@jsamr You're absolutely right, that does change the output, my mistake.
However, instead of reverting, how about this: (see my current branch)
echo* and failAndExit functions accept multiple arguments. Implode each argument with a newline, and prepend with $scriptName. This result is piped to indent. This makes the output the same as before.
Personally I prefer:
> echogood "string" \\
> "string2" \\
> "string3"

over:

> echogood "string\\nstring2\\nstring3"

I find it to be more readable. But I'll revert the change if you would like to keep the old syntax. Let me know!

**Comment by jsamr** [comment]
@SibrenVasse Yes, that would be much better, thanks :-)
Even better:

``` bash
echogood "string1" \
         "string2" \
         "string3"
```

**Edit**: Just to be very clear, I prefer your style over the existing one!

**Comment by SibrenVasse** [comment]
@jsamr Great!
As for the indentation: the style currently used in the branch is the style dictated by shfmt... Do you want to deviate from that?

**Comment by jsamr** [comment]
@SibrenVasse No I'm OK with `shfmt` ; I think I mistakenly committed an indentation change with VSCode, which is my new editor, and had lost the habit of using `shfmt`. So please feel free to re-run `shfmt` and commit!

**Comment by SibrenVasse** [comment]
@jsamr I'm sorry, what I meant was:
shfmt changed this (same as your comment)
image
to this:
image

**Comment by jsamr** [comment]
@SibrenVasse Oh I see ; yeah never mind, let's follow `shfmt` and don't bother!

**Comment by jsamr** [comment]
@SibrenVasse Since I'm not going to work on 3.3 before spring, I'm going to fast release this patch.

**Pull request merged**